internals/overlord/checkstate/manager.go

```diff
@@ -349,10 +375,75 @@ type CheckInfo struct {
 type CheckStatus string
 
 const (
-	CheckStatusUp   CheckStatus = "up"
-	CheckStatusDown CheckStatus = "down"
+	CheckStatusUp       CheckStatus = "up"
+	CheckStatusDown     CheckStatus = "down"
+	CheckStatusInactive CheckStatus = "inactive"
 )
 
 type checker interface {
 	check(ctx context.Context) error
 }
+
+// StartChecks starts the specified checks by creating a new perform-checks
+// change for each check that is in the current plan and not already running.
+func (m *CheckManager) StartChecks(currentPlan *plan.Plan, checks []string) ([]string, error) {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// If any check specified is not in the plan, return an error.
+	for _, name := range checks {
+		if _, ok := currentPlan.Checks[name]; !ok {
+			return nil, fmt.Errorf("cannot find check %q in plan", name)
+		}
+	}
+	var started []string
+	for _, name := range checks {
+		check := currentPlan.Checks[name] // We know this is ok because we checked it above.
+		info, ok := m.checks[name]
+		if !ok {
+			panic(fmt.Sprintf("check %s is in the plan but not known to the manager", name))
+		}
+		// If the check is already running, skip it.
```

Comment:
Let's be sure to consider what happens in the race conditions: 1) if the plan changes between fetching it and using it here, and 2) if `m.checks` changes between fetching it and creating starting it (creating the perform-check change).

It doesn't matter *too* much IMO, as long as it doesn't crash or do something weird.

In addition, I think there's a data race here: `m.checks` needs a mutex around access, in case another goroutine modifies it. There's the existing `checksLock` for that, though I haven't throught through what the scope of the lock/unlock should be here (perhaps around the whole fetch / update operation to avoid that race?).

Reply:
Think this through in detail to be sure, but e.g. it might be that instead of the panicking it should be more defensive instead (like continuing and logging what happened instead).